R/theme.R

```diff
@@ -92,30 +94,58 @@ highlight_styles <- function() {
   path_ext_remove(path_file(paths))
 }
 
-get_bootswatch_theme <- function(pkg) {
-  pkg$meta[["template"]]$bootswatch %||%
-    pkg$meta[["template"]]$params$bootswatch %||%
-    "_default"
+get_bslib_theme <- function(pkg) {
+  preset <-
+    pkg$meta[["template"]]$bslib$preset %||%
+    pkg$meta[["template"]]$params$bslib$preset
+
+  if (!is.null(preset)) {
+    check_bslib_theme(preset, pkg, c("template", "bslib", "preset"))
+    return(preset)
+  }
+
+  bootswatch <-
+    pkg$meta[["template"]]$bootswatch %||%
+    pkg$meta[["template"]]$params$bootswatch
+
+  if (!is.null(bootswatch)) {
+    check_bslib_theme(bootswatch, pkg, c("template", "bootswatch"))
+    return(bootswatch)
+  }
+
+  "default"
 }
 
-check_bootswatch_theme <- function(bootswatch_theme, bs_version, pkg) {
-  if (bootswatch_theme == "_default") {
-    NULL
-  } else if (bootswatch_theme %in% bslib::bootswatch_themes(bs_version)) {
-    bootswatch_theme
-  } else {
-    cli::cli_abort(c(
-      sprintf(
-        "Can't find Bootswatch theme {.val %s} ({.field %s}) for Bootstrap version {.val %s} ({.field %s}).",
-        bootswatch_theme,
-        pkgdown_field(pkg, c("template", "bootswatch")),
-        bs_version,
-        pkgdown_field(pkg, c("template", "bootstrap"))
-      ),
-      x = "Edit settings in {.file {pkgdown_config_relpath(pkg)}}"
-    ), call = caller_env())
+check_bslib_theme <- function(theme, pkg, field = c("template", "bootswatch"), bs_version = pkg$bs_version) {
+  bslib_themes <- c(
+    bslib::bootswatch_themes(bs_version),
+    if (packageVersion("bslib") >= "0.5.1") {
+      # bslib >= 0.5.1 provides built-in theme presets
+      bslib::builtin_themes(bs_version)
+    },
+    # bs_theme() recognizes both below as bare bootstrap
+    "default",
+    "bootstrap"
+  )
+
+  if (theme == "_default") {
+    return("default")
+  }
```

Comment:
```suggestion
  if (theme %in% c("_default", "default")) {
    return("default")
  }
```

And then move to the start of the function? That seems like an easy simple case to handle up front.